parseAndPopulate/capability.py

```diff
@@ -200,18 +213,24 @@ def initialize(self, impl):
                                        'vendor': impl['vendor'],
                                        'os': impl['os-type']})
 
-    def parse_and_dump_sdo(self):
-        repo = None
+    def parse_and_dump_sdo(self, repo=None):
+        """
+        If modules were sent via API, content of prepare-sdo.json file is parsed and modules are loaded from Git repository.
+        Otherwise, all the .yang files in the directory are parsed.
+
+        :param repo     Git repository which contains .yang files
+        """
+        repo = repo
         branch = None
         if self.api:
             LOGGER.debug('Parsing sdo files sent via API')
-            with open(self.json_dir + '/prepare-sdo.json', 'r') as f:
+            with open('{}/prepare-sdo.json'.format(self.json_dir), 'r') as f:
                 sdos_json = json.load(f)
             sdos_list = sdos_json['modules']['module']
             for sdo in sdos_list:
                 file_name = unicodedata.normalize('NFKD', sdo['source-file']['path'].split('/')[-1]) \
                     .encode('ascii', 'ignore')
-                LOGGER.info('Parsing sdo file sent via API {}'.format(file_name))
+                LOGGER.info('Parsing sdo file sent via API {}'.format(file_name.decode('utf-8', 'strict')))
```

Comment:
did this cause troubles before with no decode?

Reply:
The message was in the following format:
Parsing sdo file sent via API b'name-of-module.yang'